src/main.c

```diff
@@ -317,9 +322,13 @@ int main(int argc, char *argv[])
 	printf("\n");
 
 	// set up system state
-	// 512K of RAM
+	// RAM sizes come from config, default 2 Meg for each kind of memory
 	int i;
-	if ((i = state_init(2048*1024, 2048*1024)) != STATE_E_OK) {
+	int base_memory = fbc_get_int("memory", "base_memory");
+	int extended_memory = fbc_get_int("memory", "extended_memory");
```

Comment:
Extended memory - available as either a 512K or a 2MB RAM card, or the EIA Combo Card could be configured as 512K, 1MB or 1.5MB. I've no idea what granularity the OS uses to check for RAM... might be better to stick to 512K increments here too.